app/main.py

```diff
@@ -1,6 +1,22 @@
-from typing import Callable
+from functools import wraps
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        # Ініціалізуємо кеш, якщо він не існує як атрибут wrapper
+        if not hasattr(wrapper, "_cache"):
+            wrapper._cache = {}
+
+        # Перевіряємо, чи є результат для цих аргументів у кеші
+        if args in wrapper._cache:
```

Comment:
Using `args` as a key in the cache dictionary can lead to issues if any of the arguments are mutable (e.g., lists or dictionaries). Mutable objects are not hashable and cannot be used as dictionary keys. Consider converting `args` to a hashable type, such as a tuple of tuples, if the arguments might include mutable types.